registry/handlers/app.go

```diff
@@ -211,44 +212,60 @@ func NewApp(ctx context.Context, config *configuration.Configuration) *App {
 		options = append(options, storage.EnableRedirect)
 	}
 
-	// configure storage caches
-	if cc, ok := config.Storage["cache"]; ok {
-		v, ok := cc["blobdescriptor"]
-		if !ok {
-			// Backwards compatible: "layerinfo" == "blobdescriptor"
-			v = cc["layerinfo"]
-		}
-
-		switch v {
-		case "redis":
-			if app.redis == nil {
-				panic("redis configuration required to use for layerinfo cache")
-			}
-			cacheProvider := rediscache.NewRedisBlobDescriptorCacheProvider(app.redis)
-			localOptions := append(options, storage.BlobDescriptorCacheProvider(cacheProvider))
-			app.registry, err = storage.NewRegistry(app, app.driver, localOptions...)
-			if err != nil {
-				panic("could not create registry: " + err.Error())
-			}
-			ctxu.GetLogger(app).Infof("using redis blob descriptor cache")
+	if config.Metadata.Enabled {
+		var metadataService metadata.MetadataService
+		switch config.Metadata.Type {
 		case "inmemory":
-			cacheProvider := memorycache.NewInMemoryBlobDescriptorCacheProvider()
-			localOptions := append(options, storage.BlobDescriptorCacheProvider(cacheProvider))
-			app.registry, err = storage.NewRegistry(app, app.driver, localOptions...)
-			if err != nil {
-				panic("could not create registry: " + err.Error())
-			}
-			ctxu.GetLogger(app).Infof("using inmemory blob descriptor cache")
+			metadataService = inmemory.NewMetadataService()
 		default:
-			if v != "" {
-				ctxu.GetLogger(app).Warnf("unknown cache type %q, caching disabled", config.Storage["cache"])
+			panic("metadata store type must be 'inmemory'")
```

Comment:
A generic SQL implementation.  The reference implementation (and the only one we will write/support) is postgres